src/net.h

```diff
@@ -1058,7 +1058,8 @@ class CConnman
         uint64_t nMaxOutboundLimit = 0;
         int64_t m_peer_connect_timeout = DEFAULT_PEER_CONNECT_TIMEOUT;
         std::vector<std::string> vSeedNodes;
-        std::vector<NetWhitelistPermissions> vWhitelistedRange;
+        std::vector<NetWhitelistPermissions> vWhitelistedRangeIncoming;
```

Comment:
622a829ca1473814928ba32f003480b2382701a2

what's the benefit of having two separate vectors?

Reply:
You can specify different permissions for each connection direction.

Reply:
perhaps two separate lists (or a mapping to bool) could be added when it's needed, and not now that it's unused?

Reply:
I don't think it's unused because you can specify `-whitelist=noban,in@ipaddr -whitelist=mempool,out@sameipaddr`.

Reply:
good point.

Reply:
well, actually....you still can track it with a map<addr, bool>, no?

Reply:
yes, makes sense.

Reply:
I think so, but we would have to iterate the whole vector everytime we want to get specific permissions (only outbound or inbound)?

Reply:
But we don't do this much anyway?
I think the code could be easily modified for this change, you'd only have to pass addr direction to `AddWhitelistPermissionFlags. I won't insist though.